### ISSUE TITLE
missing File.readlink

### ISSUE THREAD

**Issue by kostya** [crystal-lang/crystal#3284]
for linux it implemented like this:

``` crystal
    lib LibC
      fun readlink(pathname : UInt8*, buf : UInt8*, bufsiz : LibC::SizeT) : LibC::SizeT
    end

    def File.readlink(path)
      buf = uninitialized UInt8[1024]
      size = LibC.readlink(path.to_unsafe, buf.to_unsafe, 1024).to_i32

      if size == -1
        raise Errno.new("readlink")
      elsif size > 1024
        raise "buffer too small"
      else
        return String.new(buf.to_unsafe, size)
      end
  end
```

i don't know how to correct add it because https://github.com/crystal-lang/crystal/tree/master/src/lib_c dir is quite unclear

**Comment by asterite** [comment]
We use https://github.com/ysbaddaden/posix to generate the C function. I personally never used it, but it should work. If you manage to use it please send a PR, otherwise I'll try to implement this next week anyway :-)

I also see in Ruby that they use a buffer of size 100, then if it doesn't fit they double the capacity and so on. Maybe we can do something similar.

**Comment by jaydorsey** [comment]
This looks like it's been introduced in 0.30.0. The documentation is [here](https://crystal-lang.org/api/0.30.0/File.html#readlink(path):String-class-method)

**Comment by asterite** [comment]
Closed by #7858
